src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -724,6 +724,15 @@ bool C2Compiler::is_intrinsic_supported(vmIntrinsics::ID id) {
   case vmIntrinsics::_storeFence:
   case vmIntrinsics::_storeStoreFence:
   case vmIntrinsics::_fullFence:
+  case vmIntrinsics::_isConstantExpressionZ:
```

Comment:
Move this closer to `vmIntrinsics::_isCompileConstant:`, if not outright replace it?

Reply:
I have replaced `MHImpl::isCompileConstant` with the new one.